src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2554,7 +2630,6 @@ private BoundStatement BindReturn(ReturnStatementSyntax syntax, DiagnosticBag di
             {
                 BindValueKind requiredValueKind = GetRequiredReturnValueKind(refKind);
                 arg = BindValue(expressionSyntax, diagnostics, requiredValueKind);
-                arg = ValidateEscape(arg, Binder.ExternalScope, refKind != RefKind.None, diagnostics);
```

Comment:
>ValidateEscape [](start = 22, length = 14)

nit: what was the motivation to move this somewhere below?

Reply:
We may have a switch expression here that has not been converted, so we do not have the correct expressions for its arms and cannot compute its escape scope yet.

---
In reply to: [301785766](https://github.com/dotnet/roslyn/pull/37052#discussion_r301785766) [](ancestors = 301785766)